src/cms/views/pages/page_actions.py

```diff
@@ -112,46 +111,23 @@ def delete_page(request, page_id, region_slug, language_code):
     )
 
 
-@login_required
-def copy_short_url(request, page_id, region_slug, language_code):
+def expand_short_url(request, short_url_id):
     """
-    Creates short url of page and copies url to clipboard.
+    Searches for a page with requested short_url_id and redirects to that page.
     """
-    region = Region.get_current_region(request)
-    page = get_object_or_404(region.pages, id=page_id)
-    page_translation = page.get_translation(language_code)
-
-    short_url = f"{request.scheme}://{request.get_host()}/{page_translation.short_url}"
-    try:
-        pyperclip.copy(short_url)
-        messages.success(
-            request,
-            _("URL '{}' was successfully copied to clipboard.".format(short_url)),
-        )
-    # pylint: disable=broad-except
-    except Exception as e:
-        logger.exception(e)
-        messages.warning(
-            request,
-            _("URL '{}' could not be copied to clipboard.".format(short_url)),
-        )
+    queryset = PageTranslation.objects.filter(short_url_id=short_url_id)
+    page_translation = queryset.first()
 
-    return redirect(
-        "pages",
-        **{
-            "region_slug": region_slug,
-            "language_code": language_code,
-        },
-    )
+    if page_translation:
+        return redirect(WEBAPP_URL + page_translation.get_absolute_url())
+    return HttpResponseNotFound("<h1>Page not found</h1>")
 
 
-@login_required
-def expand_short_url(request, short_url_id):
+def expand_page_translation_id(request, short_url_id):
     """
-    Searches for a page with requested short_url_id and redirects to that page.
+    Searches for a page translation with corresponding ID and redirects browser to web app
     """
-    queryset = PageTranslation.objects.filter(short_url_id=short_url_id)
-    page_translation = queryset.first()
+    page_translation = PageTranslation.objects.get(id=short_url_id)
```

Comment:
Needs a check if a page is published.

Reply:
Also, there might be a newer revision of the page translation...
And we have to think about what to do on a 404... would we show the 404 of the cms, or redirect to the 404 in the webapp?

Reply:
Okay, we actually should redirect to the newest version (slug), for sure. About the 404: good question.

Reply:
I think the webapp has no generic 404 page, if we redirect e.g. to https://integreat.app/@@redux-first-router/NOT_FOUND, it shows `NetworkError when attempting to fetch resource.` which is a bit misleading. And when we redirect to https://integreat.app/404, it says `Sorry, but this city could not be found.`, which is also not entirely correct. So I propose to keep it as it is and show the error message of the cms...

Reply:
I'm now redirecting to the latest public revision. Should do the trick, right?

Reply:
In this case the check is not required anymore :roll_eyes: